### PR TITLE
New attribution methods tutorials and bug fixes

### PR DESCRIPTION
**Pull request by AntoninPoche**
This PR updates the attribution methods tutorials, fixes bugs in rise and smoothgrad attribution methods and in metrics plots, and correct some typo.

## Tutorials

The tutorials are google colab notebooks hosted on google drive to reduce the size of the git and allow an esay usage of gpus. Therefore, 73e83ab mainly updates tutorials links.

Also, there were no links to the github / methods source in the documentation, there were also added in 73e83ab. As well as some papers links for documentations without quotes (containing the paper link themselves).

#### Rise

The `Rise` method did not support not square images. It was corrected in eb494a5. The possibility to have different `grid_size` depending on the dimension and a test to validate it was created.

#### SmoothGrad, SquareGrad and VarGrad

The `SmoothGrad` method and the two heritating methods did not support high values for their `nb_samples` parameters because it generated an out of memeory error. It was corrected in 491a78a.
The implementation was making batches on the `inputs` and not on the samples (perturbed inputs) generated by the methods. As such, elements of shapes `(batch_size, nb_samples, h, w, c)` were generated.
The fix make batches of `(batch_size, h, w, c)` perturbed inputs and iterates on each input separately.

As a consequence, the methods `_get_noisy_mask` and `_apply_noise` were deleted as they only represented a line afterward. The corresponding test was also deleted. 

To verify that the new implementation did not destroy the method, empirical tests were effectuated afterward and the results correponds.

#### Metric plots

Both `barplot` and `fidelity_curves` functions from `xplique/plots/metrics.py` did not support when only one method was given respectively in the `scores` or `detailed_scores` parameters. It generated a division by zero.

This bug was corrected in d96f366.

## Typo

In 39c631b few elements were modified:
- `README.md`: 
  - update callable link
  - Correct the paper link for GradientInput (it was the DeepLift paper)
- `CONTRIBUTING.md`: `pip install -e` -> `pip install -e .`
- `docs/api/attributions/rise.md`: There was a problem in the formula, the sum starts at 1 and not 0.
- `xplique/plots/image.py`: typo, a space was forgotten due to line return.

Furthermore, the order of attribution methods in the documentation was set to the alphabetic order in 978581d.

### CODE REVIEW

File: xplique/plots/metrics.py [hunk 1]
```diff
@@ -70,8 +70,11 @@ def barplot(
             # methods_color is a string linking to a cmap
             cmap = matplotlib.cm.get_cmap(methods_colors)
 
-        methods_colors = {methods[i]: cmap(i / (len(methods) - 1))
-                          for i in range(len(methods))}
+        if len(methods) == 1:
```
Comment (fel-thomas) [inline xplique/plots/metrics.py:73]: if methods is a base explainer it has no len ^^
Also if len(methods) == 1 then the second line will work ;)

File: xplique/plots/metrics.py [hunk 2]
```diff
@@ -140,8 +143,11 @@ def fidelity_curves(
...
```
Comment (fel-thomas) [inline xplique/plots/metrics.py:146]: same here

File: docs/api/attributions/rise.md [hunk 3]
```diff
@@ -16,7 +16,7 @@ obtaining the corresponding outputs to deduce critical areas.
 with $S_c$ the class score **after softmax**, $x$ an input, and $m  \sim \mathcal{M}$ a mask (not
  binary) the RISE importance map $\phi$ is defined as :
 
-$$ \phi_i = \frac{1}{\mathbb{E}(\mathcal{M}) N} \sum_{i=0}^N S_c(x \odot m_i) m_i $$
+$$ \phi_i = \frac{1}{\mathbb{E}(\mathcal{M}) N} \sum_{i=1}^N S_c(x \odot m_i) m_i $$
```
Comment (fel-thomas) [inline docs/api/attributions/rise.md:19]: 👍

File: docs/api/attributions/deconvnet.md [hunk 4]
```diff
@@ -12,10 +16,11 @@ method = DeconvNet(model)
...
 ```
 
+
 ## Notebooks
 
 - [**Attribution Methods**: Getting started](https://colab.research.google.com/drive
-/1XproaVxXjO9nrBSyyy7BuKJ1vy21iHs2) <sub> [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/1XproaVxXjO9nrBSyyy7BuKJ1vy21iHs2) </sub>
-- [**DeconvNet**: Going Further](https://colab.research.google.com/drive/1qBxwsMILPvQs3WLLcX_hRb3kzTSI4rkz) <sub> [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/19oLUjmvrBIMTmNKXcJNbB6pJvkfutLEb) </sub>
+/1XproaVxXjO9nrBSyyy7BuKJ1vy21iHs2)
+- [**DeconvNet**: Going Further](https://colab.research.google.com/drive/19eB3uwAtCKZgkoWtMzrF0LTJ-htF_KE7?authuser=1)
```
Comment (fel-thomas) [inline docs/api/attributions/deconvnet.md:24]: remove auth user

File: docs/api/attributions/grad_cam.md [hunk 5]
```diff
@@ -1,5 +1,7 @@
 # Grad-CAM
 
+<sub><img src="https://upload.wikimedia.org/wikipedia/commons/d/d0/Google_Colaboratory_SVG_Logo.svg" width="20"></sub>[ View colab tutorial](https://colab.research.google.com/drive/1nsB7xdQbU0zeYQ1-aB_D-M67-RAnvt4X?authuser=1) | <sub><img src="https://github.githubassets.com/images/modules/logos_page/GitHub-Mark.png" width="20"></sub>[ View source](https://github.com/deel-ai/xplique/blob/master/xplique/attributions/grad_cam.py)
```
Comment (fel-thomas) [inline docs/api/attributions/grad_cam.md:3]: auth user

File: docs/api/attributions/grad_cam.md [hunk 6]
```diff
@@ -38,8 +40,8 @@ explanations = method.explain(images, labels)
 ## Notebooks
 
 - [**Attribution Methods**: Getting started](https://colab.research.google.com/drive
-/1XproaVxXjO9nrBSyyy7BuKJ1vy21iHs2) <sub> [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/1XproaVxXjO9nrBSyyy7BuKJ1vy21iHs2) </sub>
-- [**GradCam**: Going Further](https://colab.research.google.com/drive/1wJmPK3HZ1Uw1rBbeo8DtmmaRZUZ1Cs-i) <sub> [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/1wJmPK3HZ1Uw1rBbeo8DtmmaRZUZ1Cs-i) </sub>
+/1XproaVxXjO9nrBSyyy7BuKJ1vy21iHs2)
+- [**GradCam**: Going Further](https://colab.research.google.com/drive/1nsB7xdQbU0zeYQ1-aB_D-M67-RAnvt4X?authuser=1)
```
Comment (fel-thomas) [inline docs/api/attributions/grad_cam.md:44]: I suggest we stay consistent with "GradCAM"

File: docs/api/attributions/grad_cam_pp.md [hunk 7]
```diff
@@ -1,5 +1,7 @@
 # Grad-CAM++
 
+<sub><img src="https://upload.wikimedia.org/wikipedia/commons/d/d0/Google_Colaboratory_SVG_Logo.svg" width="20"></sub>[ View colab tutorial](https://colab.research.google.com/drive/1nsB7xdQbU0zeYQ1-aB_D-M67-RAnvt4X?authuser=1) | <sub><img src="https://github.githubassets.com/images/modules/logos_page/GitHub-Mark.png" width="20"></sub>[ View source](https://github.com/deel-ai/xplique/blob/master/xplique/attributions/grad_cam_pp.py) | 📰 [ See paper](https://arxiv.org/abs/1710.11063)
```
Comment (fel-thomas) [inline docs/api/attributions/grad_cam_pp.md:3]: auth user

File: docs/api/attributions/grad_cam_pp.md [hunk 8]
```diff
@@ -18,8 +20,8 @@ explanations = method.explain(images, labels)
 ## Notebooks
 
 - [**Attribution Methods**: Getting started](https://colab.research.google.com/drive
-/1XproaVxXjO9nrBSyyy7BuKJ1vy21iHs2) <sub> [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/1XproaVxXjO9nrBSyyy7BuKJ1vy21iHs2) </sub>
-- [**GradCam ++**: Going Further](https://colab.research.google.com/drive/1NRzdZdwxEYhC3_0gf8VpC_bg4YQcVsnO) <sub> [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/1NRzdZdwxEYhC3_0gf8VpC_bg4YQcVsnO) </sub>
+/1XproaVxXjO9nrBSyyy7BuKJ1vy21iHs2)
+- [**GradCam ++**: Going Further](https://colab.research.google.com/drive/1nsB7xdQbU0zeYQ1-aB_D-M67-RAnvt4X?authuser=1)
```
Comment (fel-thomas) [inline docs/api/attributions/grad_cam_pp.md:24]: auth, GradCAM

File: docs/api/attributions/gradient_input.md [hunk 9]
```diff
@@ -1,5 +1,7 @@
 # Gradient $\odot$ Input
 
+<sub><img src="https://upload.wikimedia.org/wikipedia/commons/d/d0/Google_Colaboratory_SVG_Logo.svg" width="20"></sub>[ View colab tutorial](https://colab.research.google.com/drive/19eB3uwAtCKZgkoWtMzrF0LTJ-htF_KE7?authuser=1) | <sub><img src="https://github.githubassets.com/images/modules/logos_page/GitHub-Mark.png" width="20"></sub>[ View source](https://github.com/deel-ai/xplique/blob/master/xplique/attributions/deconvnet.py)
```
Comment (fel-thomas) [inline docs/api/attributions/gradient_input.md:3]: authuser

File: docs/api/attributions/gradient_input.md [hunk 10]
```diff
@@ -37,7 +39,7 @@ explanations = method.explain(images, labels)
 
 - [**Attribution Methods**: Getting started](https://colab.research.google.com/drive
 /1XproaVxXjO9nrBSyyy7BuKJ1vy21iHs2) <sub> [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/1XproaVxXjO9nrBSyyy7BuKJ1vy21iHs2) </sub>
-- [**Gradient $\odot$ Input**: Going Further](https://colab.research.google.com/drive/1pHuiggijCZ0touf2gEtdM2QsiR4oFJwS) <sub> [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/1pHuiggijCZ0touf2gEtdM2QsiR4oFJwS) </sub>
+- [**Gradient $\odot$ Input**: Going Further](https://colab.research.google.com/drive/19eB3uwAtCKZgkoWtMzrF0LTJ-htF_KE7?authuser=1) <sub> [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/19eB3uwAtCKZgkoWtMzrF0LTJ-htF_KE7?authuser=1) </sub>
```
Comment (fel-thomas) [inline docs/api/attributions/gradient_input.md:42]: auth

**Review by fel-thomas — COMMENTED** [review]
Awesome tutorials, super PR 🤘 ;) 

just a few comments

**Review by lucashervier — COMMENTED** [review]
I do not have anything to add to Thomas' comments. GG!

File: xplique/plots/metrics.py (hunk 1 above)
Comment (AntoninPoche) [reply xplique/plots/metrics.py]: corrected now

File: docs/api/attributions/grad_cam_pp.md (hunk 8 above)
Comment (lucashervier) [reply docs/api/attributions/grad_cam_pp.md:24]: This one is still forgotten :)

**Review by lucashervier — APPROVED** [review]
Think to change the GradCam to GradCAM and then let's go

**Pull request merged**